build/build_system.bzl

```diff
@@ -208,3 +212,135 @@ github_release = repository_rule(
+            continue
+
+        target_path = path.replace(strip_path + "/", "")
+        full_path = "%s/%s%s" % (KONG_VAR["BUILD_NAME"], prefix, target_path)
+
+        # use a fake output, if we are writing a directory that may collide with others
+        # nop_path = "%s-nop-farms/%s/%s/%s" % (KONG_VAR["BUILD_NAME"], strip_path, prefix, target_path)
+        # output = ctx.actions.declare_file(nop_path)
+        # ctx.actions.run_shell(
+        #     outputs = [output],
+        #     inputs = [file],
+        #     command = "(mkdir -p {t} && chmod -R +rw {t} && cp -r {s} {t}) >{f}".format(
+        #         t = full_path,
+        #         s = file.path,
+        #         f = output.path,
+        #     ),
+        # )
+        output = ctx.actions.declare_file(full_path)
+        ctx.actions.run_shell(
+            outputs = [output],
+            inputs = [file],
+            command = "cp -r %s %s" % (file.path, output.path),
+        )
+
+        outputs.append(output)
+
+        if full_path.find(".so.") >= 0 and ctx.attr.create_dynamic_library_symlink:
+            el = full_path.split(".")
+            si = el.index("so")
+            sym_paths = []
+            if len(el) > si + 2:  # has more than one part after .so like libX.so.2.3.4
+                sym_paths.append(".".join(el[:si + 2]))  # libX.so.2
+            sym_paths.append(".".join(el[:si + 1]))  # libX.so
+
+            for sym_path in sym_paths:
+                sym_output = ctx.actions.declare_symlink(sym_path)
+                ctx.actions.symlink(output = sym_output, target_path = file.basename)
+                outputs.append(sym_output)
+
+    return [DefaultInfo(files = depset(outputs + ctx.files.deps))]
+
+kong_install = rule(
+    implementation = _kong_install_impl,
+    doc = "Install files from the `src` label output to BUILD_DESTDIR",
+    attrs = {
+        "prefix": attr.string(
+            mandatory = False,
+            doc = "The relative prefix to add to target files, after KONG_VAR['BUILD_DESTDIR'], default to 'kong'",
+            default = "kong",
+        ),
+        "strip_path": attr.string(
+            mandatory = False,
+            doc = "The leading path to strip from input, default to the ./<package/<target>",
+            default = "",
+        ),
+        # "include": attr.string_list(
+        #     mandatory = False,
+        #     doc = "List of files to explictly install, take effect after exclude; full name, or exactly one '*' at beginning or end as wildcard are supported",
+        #     default = [],
+        # ),
+        # "exclude": attr.string_list(
+        #     mandatory = False,
+        #     doc = "List of directories to exclude from installation",
+        #     default = [],
+        # ),
+        "create_dynamic_library_symlink": attr.bool(
+            mandatory = False,
+            doc = "Create non versioned symlinks to the versioned so, e.g. libfoo.so -> libfoo.so.1.2.3",
```

Comment:
``` suggestion
            doc = "Create symlinks to the `.so` without any suffix (except `.so`), e.g. libfoo.so -> libfoo.so.1.2.3, libfoo.so -> libfoll.so.1.2.3.rolling",
```

Reply:
normal libraries produced (typically) by libtool doesn't have `.rolling` as suffix, so we would not include the exception
here in doc.